### DISCUSSION TITLE
Please create an initial migration when the database is being initialized from scratch

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#1872] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

It doesn't make any sense to me that I would need to manually create a migration when I am creating a brand new database. Serverpod should detect when there is no migration, and create one.

Furthermore, `serverpod generate` doesn't actually create the `definition.sql` file, because it doesn't create a migration, and it feels like it should.

In fact, I'd go so far as to say that whenever `serverpod generate` detects that the data model has changed, it should automatically create a migration.

I don't like having migrations separated from the generate process, and I don't like the idea of having to manually initialize the system the first time.

**Comment by vlidholt** [comment, marked as answer]
If you are developing your server locally, you can achieve this by always passing the `--apply-migrations` flag to the server. E.g.,

```bash
dart bin/main.dart --apply-migrations
```

It's important to separate the migrations from the generate command as migrations may need to be manually modified in some cases. They can also potentially be destructive, which would require special attention. You can also remove a set of migrations and regenerate them when needed.

One (simplified) example is that you remove a column, but later decides to add it back in. In that case having two migrations would destroy the column, but being able to remove the two migrations and generate a new one can keep the data.
